### PR TITLE
sql: Add catalog views for privileges and roles

### PR DESCRIPTION
**Pull request by jkosh44**
This commit adds a number of catalog views that present privileges and role memberships using human-readable names instead of IDs. Each view has two variants, one that presents all privileges or roles, and another that only presents privileges and roles that contain the current role.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by nrainer-materialize — APPROVED** [review]

**Review by mjibson — APPROVED** [review]

**Comment by jkosh44** [comment]
For the non-engineering folks, you can just take a look at `mz_internal.md` for descriptions of the views, and `rbac_views.slt` for some example uses.

File: doc/user/content/sql/system-catalog/mz_internal.md [hunk 1]
```diff
@@ -209,6 +209,236 @@ The `mz_sessions` table contains a row for each active session in the system.
...
+in the system on to user roles.
+
+<!-- RELATION_SPEC mz_internal.mz_show_system_privileges -->
+| Field            | Type     | Meaning                                     |
+|------------------|----------|---------------------------------------------|
+| `grantor`        | [`text`] | The role that granted the privilege.        |
+| `grantee`        | [`text`] | The role that the privilege was granted to. |
+| `privilege_type` | [`text`] | They type of privilege granted.             |
+
+### `mz_show_my_all_privileges`
```
Comment (chaas) [inline doc/user/content/sql/system-catalog/mz_internal.md:323]: nit: maybe `mz_show_all_my_privileges` sounds more logical, even if it's not technically consistent with e.g. `mz_show_my_cluster_privileges`?

**Review by chaas — APPROVED** [review]
This is great, thanks Joe!

@morsapaes for the docs, we could highlight these as part of the tutorial process.
We might also want to include in the docs how to filter these views for a specific user, e.g. how to get the privileges of a specific user.
Another thing for the docs is a query for getting all the roles a given role is a direct or indirect member of. That query is currently broken but I think a fix will go out next week https://github.com/MaterializeInc/materialize/issues/20769 and we can write it up then.

File: doc/user/content/sql/system-catalog/mz_internal.md (hunk 1 above)
Comment (jkosh44) [reply doc/user/content/sql/system-catalog/mz_internal.md:323]: Yeah, I was going back and forth on this. I'll switch it to `mz_show_all_my_privileges`.

**Comment by jkosh44** [comment]
> We might also want to include in the docs how to filter these views for a specific user, e.g. how to get the privileges of a specific user.

That specific query would be 
```SQL
SELECT * FROM mz_internal.show_<object-type>_privileges WHERE pg_has_role(<role-name>, grantee, 'USAGE');
```

> Another thing for the docs is a query for getting all the roles a given role is a direct or indirect member of.

That query would be
```SQL
SELECT * FROM mz_internal.mz_show_role_members WHERE pg_has_role(<role-name>, member, 'USAGE');
```

**Comment by jkosh44** [comment]
I'm going to go ahead and merge this. Since it's `mz_internal` we can always change it later and it's probably better to experiment with these views in staging/prod rather than staring at this PR.

**Pull request merged**